documentation/modules/oauth/con-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,452 @@
+For example:
+
+    Dev Team A can write to topics that start with x_ on any cluster
+
+    Dev Team B can read from topics that start with x_ on any cluster
+    Dev Team B can update consumer group offsets that start with x_ on any cluster
+
+    ClusterManager of my-cluster Group has full access to cluster config on my-cluster
+    ClusterManager of my-cluster Group has full access to consumer groups on my-cluster
+    ClusterManager of my-cluster Group has full access to topics on my-cluster
+
+If we take a closer look at the `Dev Team A can write ...` permission definition, we see that it combines a resource called `Topic:x_*`, scopes `Describe` and `Write`, and `Dev Team A` policy.
+If we click on the `Dev Team A` policy, we see that it matches all users that have a realm role called `Dev Team A`.
+
+Similarly, the `Dev Team B ...` permissions perform matching using the `Dev Team B` policy which also uses realm role to match allowed users - in this case those with realm role `Dev Team B`.
+The `Dev Team B ...` permissions grant users `Describe` and `Read` on `Topic:x_*`, and `Group:x_*` resources, effectively giving matching users and clients the ability to read from topics, and update the consumed offsets for topics and consumer groups that have names starting with 'x_'.
+
+## Targeting Permissions - Clients and Roles vs. Users and Groups
+
+In Keycloak, confidential clients with 'service accounts' enabled can authenticate to the server in their own name using a clientId and a secret.
+This is convenient for microservices which typically act in their own name, and not as agents of a particular user (like a web site would, for example).
+Service accounts can have roles assigned like regular users.
+They cannot, however, have groups assigned.
+As a consequence, if you want to target permissions to microservices using service accounts, you cannot use Group policies, and should instead use Role policies.
+Or, thinking about it another way, if you want to limit certain permissions only to regular user accounts where authentication with username and password is required, you can achieve that as a side effect of using the Group policies, rather than the Role policies.
+That's what we see used in `permissions` that start with 'ClusterManager'.
+Performing cluster management is usually done interactively - in person - using CLI tools.
+It makes sense to require the user to log-in, before using the resulting access token to authenticate to the Kafka Broker.
+In this case the access token represents the specific user, rather than the client application.
+
+
+## Authorization in Action Using CLI Clients
+
+A note of caution - this example uses the latest version of Keycloak, and relies on bugfixes of issues in some older versions.
+Let's make sure the Authorization rules have been properly imported when the Keycloak was started.
```

Comment:
```suggestion
Before trying any new authorization configuration, make sure that authorization rules have been properly imported when Keycloak starts.
```

Reply:
Can we drop this section then? Concerned about future-proofing the content

Reply:
Ok. Let's drop it then.

Reply:
Or what if we just do it like this:
```suggestion
Make sure that authorization rules have been properly imported.
```